bgzip.c

```diff
@@ -198,6 +203,9 @@ int main(int argc, char **argv)
 
         if ( index ) bgzf_index_build_init(fp);
         buffer = malloc(WINDOW_SIZE);
+#ifdef _WIN32
+        _setmode(f_src, O_BINARY);
+#endif
```

Comment:
Reread my comment.  It's talking about changing _bgzip.c_ to use `hopen` instead of `open`.  It does not propose altering the implementation of `hopen` itself.

I changed bgzip's BGZF files from `open` to (effectively) `hopen` in b8204c12e1a474cf4b3ed845e0323f71a12c0cda during the PR you mention, which is linked in my previous comment.  I would have rather liked to change its other files too as it simplifies and regularises the code, but I had not analysed the effects that would have on Windows, as I noted at the time.

> The change was made for a reason (it fixes bugs).

Apparently you have now analysed the effects, though that wasn't apparent from the commit messages.  Grand; and I guess it gives `bgzip -i` a chance to have the right offsets.

So the maintenance question is about the risks of a change that affects all platforms while regularising and simplifying the code, versus one that adds the maintenance burden of `#ifdef` spaghetti.